### ISSUE TITLE
Startup crash on Python 3.5.1

### ISSUE THREAD

**Issue by pythoneerd** [PyCQA/pylint#2632]
### Steps to reproduce
1. Install pylint >= 2.2
2. Run pylint

### Current behavior
> (spyder) C:\Users\user\pyvenvs\spyder>pylint
Problem importing module async.py: cannot import name 'Type'
Problem importing module base.py: cannot import name 'Type'
Problem importing module classes.py: cannot import name 'Type'
Problem importing module design_analysis.py: cannot import name 'Type'
Problem importing module exceptions.py: cannot import name 'Type'
Problem importing module format.py: cannot import name 'Type'
Problem importing module imports.py: cannot import name 'Type'
Problem importing module logging.py: cannot import name 'Type'
Problem importing module newstyle.py: cannot import name 'Type'
Problem importing module python3.py: cannot import name 'Type'
Problem importing module refactoring.py: cannot import name 'Type'
Problem importing module spelling.py: cannot import name 'Type'
Problem importing module stdlib.py: cannot import name 'Type'
Problem importing module strings.py: cannot import name 'Type'
Problem importing module typecheck.py: cannot import name 'Type'
Problem importing module utils.py: cannot import name 'Type'
Problem importing module variables.py: cannot import name 'Type'
Traceback (most recent call last):
  File "c:\program files\python35\lib\runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\program files\python35\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\user\pyvenvs\spyder\Scripts\pylint.exe\__main__.py", line 9, in <module>
  File "c:\users\user\pyvenvs\spyder\lib\site-packages\pylint\__init__.py", line 20, in run_pylint
    Run(sys.argv[1:])
  File "c:\users\user\pyvenvs\spyder\lib\site-packages\pylint\lint.py", line 1557, in __init__
    linter.enable("c-extension-no-member")
  File "c:\users\user\pyvenvs\spyder\lib\site-packages\pylint\utils.py",line 323, in enable
    msgid, enable=True, scope=scope, line=line, ignore_unknown=ignore_unknown
  File "c:\users\user\pyvenvs\spyder\lib\site-packages\pylint\utils.py",line 366, in _set_msg_status
    msg = self.msgs_store.get_message_definition(msgid)
  File "c:\users\user\pyvenvs\spyder\lib\site-packages\pylint\utils.py",line 989, in get_message_definition
    msgid_or_symbol=msgid_or_symbol
pylint.exceptions.UnknownMessageError: No such message id c-extension-no-member

### pylint --version output
Version output is not available (crashes too).
pip reports astroid version 2.1.0
pylint versions as stated above.

### suspected problem
Trying to import `Type` from the typing module, which is not supported under 3.5.1, for example in checkers/utils.py

Support for typing is incomplete in 3.5.1, see this IDLE session:

`Python 3.5.1 (v3.5.1:37a07cee5969, Dec  6 2015, 01:54:25) [MSC v.1900 64 bit (AMD64)] on win32`
`Type "copyright", "credits" or "license()" for more information.`
`>>> import typing`
`>>> from typing import Optional`
`>>> from typing import Iterable`
`>>> from typing import Tuple`
`>>> from typing import Callable`
`>>> from typing import Set`
`>>> from typing import Union`
`>>> from typing import Match`
`>>> from typing import Dict`
`>>> from typing import List`
`>>> from typing import Type`
`Traceback (most recent call last):`
`  File "<pyshell#10>", line 1, in <module>`
`    from typing import Type`
`ImportError: cannot import name 'Type'`
`>>> `

**Comment by PCManticore** [comment]
Thanks! Should be fixed with latest master. We're testing 3.5 in our CI, but for some reason 3.5.5 has that attribute, while 3.5.1 doesn't.

**Comment by pythoneerd** [comment]
Wow, that was incredibly fast! Thank you for fixing this.
The whole typing thing was brand new in 3.5, and this particular class was added in [this commit](https://github.com/python/cpython/commit/eb9aca3c071aab925831a389ab26816c170dc159) on 2016-05-25, appearing first in 3.5.2.rc1.
I tracked this down because this is not my first time to run into typing-related problems. Maybe i should abandon 3.5.1.
